myscript.js

```diff
@@ -0,0 +1,198 @@
+
+const myUrl =  "http://localhost:5050/getFibonacciResults";
+let fibonacciUrl = "http://localhost:5050/fibonacci/";
+
+let myArr = [];
+let myIndex = document.getElementById("index");
+let myY = document.getElementById('y');
+let myMsg = document.getElementById('myMsg');
+let container = document.getElementById('containerDiv');
+let myCheckBox = document.getElementById('saveCalc');
+let isCalculationSaved; 
+let aListDropdown = document.getElementById('aListDropdown');
+
+window.addEventListener('load', init);
+
+aListDropdown.addEventListener('click', (e)=> {
+    container.innerHTML = "";
+    if (e.target.innerText  === "Number Asc") {   
+        spinnerOn();  
+        updateMyArr("Number Asc");
+    }
+    else if (e.target.innerText  === "Number Desc") {
+        spinnerOn();  
+        updateMyArr("Number Desc");
+    }
+    else if (e.target.innerText  === "Date Asc") {
+        spinnerOn();  
+        updateMyArr("Date Asc");
+    }
+    else if(e.target.innerText  === "Date Desc") {
+        spinnerOn();  
+        updateMyArr("Date Desc");
+    }
+});
+
+
+function spinnerOn() {
+    spinner.classList.add("spinner-border");
+    setTimeout(()=>spinner.classList.remove("spinner-border"),2000);
+}
+
+async function init(){
+    const response = await fetch(myUrl);
+    const data = await response.json();
+    myArr = data.results;
+    BuildResults();
+}
+
+async function updateMyArr(sortingOption){
+    const response = await fetch(myUrl);
+    const data = await response.json();
+    myArr = sorting(sortingOption);
+    BuildResults();
+
+}
+
+let serverMsg = "";
+async function serverWriteData(url){
+    const response = await fetch(url);
+    try {
+        if (!response.ok) {
+            text = await response.text();
+            serverMsg = text;
```

Comment:
You should throw the error here and actually do the displaying in the .catch;

`throw new Error(text)`